test/e2e/autoscaling/dns_autoscaling.go

```diff
@@ -237,13 +237,13 @@ func getExpectReplicasFuncLinear(c clientset.Interface, params *DNSParamsLinear)
 	return func(c clientset.Interface) int {
 		var replicasFromNodes float64
 		var replicasFromCores float64
-		nodes, err := e2enode.GetReadySchedulableNodes(c)
```

Comment:
is it possible to use this function instead? 
https://github.com/kubernetes/kubernetes/blob/e1f84d52b7d1ef8c1014d62095302af2dee838c5/test/e2e/framework/node/resource.go#L380-L384

Reply:
That was probably the one I was looking for :-) But I think the proliferation of these functions is not very sustainable.  I figured it was better to be very explicit about which nodes we wanted which we can do more easily with this sort of interface IMO.

If we're not sure though, I can split up this PR into two - one that just switches to GetReadyNodesIncludingTainted (fixing the test issue), and one that introduces the fluent interface (as I expect that might get more discussion).

Reply:
nah, is ok, bad wording, I wanted to mean if those were equivalent ...

Reply:
Thanks - I actually don't mind splitting it - particularly now you've found the function I was looking for :-)  I'm pretty sure I would have called it out if someone sent me this PR - "first fix the test flake, then refactor separately" or something like that!

If we both are pretty ambivalent we can see if anyone else chimes in though...

Reply:
the problems with the e2e folder is that any refactor is very dangerous ... and sometimes we don't notice something has broken after some days or weeks before :shrug:  

>  "first fix the test flake, then refactor separately" 

that seems the wise thing to do :+1:

Reply:
Thanks - split out the refactor in #98070; this is just the test fix now - thanks again for finding the right method!